spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/sql/R2dbcCloudSqlEnvironmentPostProcessor.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spring.autoconfigure.sql;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.bind.Binder;
+import org.springframework.boot.context.properties.source.ConfigurationPropertySources;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
+
+/**
+ * Builds connection string for Cloud SQL through Spring R2DBC by requiring only a
+ * database and instance connection name.
+ *
+ */
+public class R2dbcCloudSqlEnvironmentPostProcessor implements EnvironmentPostProcessor {
```

Comment:
Because there is a lot shared code, would it make sense to just enhance the `CloudSqlEnvironmentPostProcessor` instead of introducing another `EnvrionmentPostProcessor`?

Reply:
Ah that is a good idea! I was initially hesitant about putting them both in `CloudSqlEnvironmentPostProcessor` as it would result in extra `if-else` statements to determine whether it is the r2dbc path or the jdbc path. However, given the presence of so much shared code it might be worth it! I'll try this out.